// office-ui-fabric-react/lib-amd/components/Button/CompoundButton/CompoundButton.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "../BaseButton", "./CompoundButton.scss"], function (require, exports, BaseButton_1, CompoundButton_scss_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var CompoundButton = (function (_super) {
        __extends(CompoundButton, _super);
        function CompoundButton() {
            var _this = _super !== null && _super.apply(this, arguments) || this;
            _this.classNames = {
                base: 'ms-Button',
                variant: 'ms-Button--compound',
                description: CompoundButton_scss_1.default.description,
                flexContainer: CompoundButton_scss_1.default.flexContainer,
                icon: null,
                isDisabled: CompoundButton_scss_1.default.isDisabled,
                isEnabled: CompoundButton_scss_1.default.isEnabled,
                label: CompoundButton_scss_1.default.label,
                root: CompoundButton_scss_1.default.root
            };
            return _this;
        }
        return CompoundButton;
    }(BaseButton_1.BaseButton));
    exports.CompoundButton = CompoundButton;
});
