// office-ui-fabric-react/lib/demo/AppDefinition.js
"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var example_app_base_1 = require("@uifabric/example-app-base");
var DetailsList_Basic_Example_1 = require("../components/DetailsList/examples/DetailsList.Basic.Example");
exports.AppDefinition = {
    appTitle: 'Fabric - React',
    testPages: [
        {
            component: DetailsList_Basic_Example_1.DetailsListBasicExample,
            key: 'DetailsListBasicExample',
            name: 'DetailsListBasicExample',
            url: '#/tests/detailslistbasicexample'
        }
    ],
    examplePages: [
        {
            links: [
                {
                    getComponent: function (cb) { return cb(require('../components/Breadcrumb/BreadcrumbPage').BreadcrumbPage); },
                    key: 'Breadcrumb',
                    name: 'Breadcrumb',
                    url: '#/examples/breadcrumb'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Button/ButtonPage').ButtonPage); },
                    key: 'Button',
                    name: 'Button',
                    url: '#/examples/button'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Calendar/CalendarPage').CalendarPage); },
                    key: 'Calendar',
                    name: 'Calendar',
                    url: '#/examples/calendar'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Callout/CalloutPage').CalloutPage); },
                    key: 'Callout',
                    name: 'Callout',
                    url: '#/examples/callout'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Checkbox/CheckboxPage').CheckboxPage); },
                    key: 'Checkbox',
                    name: 'Checkbox',
                    url: '#/examples/checkbox'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/ChoiceGroup/ChoiceGroupPage').ChoiceGroupPage); },
                    key: 'ChoiceGroup',
                    name: 'ChoiceGroup',
                    url: '#/examples/choicegroup'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/CommandBar/CommandBarPage').CommandBarPage); },
                    key: 'CommandBar',
                    name: 'CommandBar',
                    url: '#/examples/commandbar'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/ContextualMenu/ContextualMenuPage').ContextualMenuPage); },
                    key: 'ContextualMenu',
                    name: 'ContextualMenu',
                    url: '#/examples/contextmenu'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/DatePicker/DatePickerPage').DatePickerPage); },
                    key: 'DatePicker',
                    name: 'DatePicker',
                    url: '#/examples/datepicker'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/DetailsList/DetailsListPage').DetailsListPage); },
                    key: 'DetailsList',
                    name: 'DetailsList',
                    url: '#/examples/detailslist'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Dialog/DialogPage').DialogPage); },
                    key: 'Dialog',
                    name: 'Dialog',
                    url: '#/examples/dialog'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/DocumentCard/DocumentCardPage').DocumentCardPage); },
                    key: 'DocumentCard',
                    name: 'DocumentCard',
                    url: '#/examples/documentcard'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Dropdown/DropdownPage').DropdownPage); },
                    key: 'Dropdown',
                    name: 'Dropdown',
                    url: '#/examples/dropdown'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Facepile/FacepilePage').FacepilePage); },
                    key: 'Facepile',
                    name: 'Facepile',
                    url: '#/examples/facepile'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Icon/IconPage').IconPage); },
                    key: 'Icon',
                    name: 'Icon',
                    url: '#/examples/icon'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Label/LabelPage').LabelPage); },
                    key: 'Label',
                    name: 'Label',
                    url: '#/examples/label'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Link/LinkPage').LinkPage); },
                    key: 'Link',
                    name: 'Link',
                    url: '#/examples/link'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/List/ListPage').ListPage); },
                    key: 'List',
                    name: 'List',
                    url: '#/examples/list'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/MessageBar/MessageBarPage').MessageBarPage); },
                    key: 'MessageBar',
                    name: 'MessageBar',
                    url: '#/examples/messagebar'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Modal/ModalPage').ModalPage); },
                    key: 'Modal',
                    name: 'Modal',
                    url: '#/examples/modal'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Overlay/OverlayPage').OverlayPage); },
                    key: 'Overlay',
                    name: 'Overlay',
                    url: '#/examples/overlay'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Panel/PanelPage').PanelPage); },
                    key: 'Panel',
                    name: 'Panel',
                    url: '#/examples/panel'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/pickers/PickersPage').PickersPage); },
                    key: 'Pickers',
                    name: 'Pickers',
                    url: '#/examples/pickers'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/pickers/PeoplePicker/PeoplePickerPage').PeoplePickerPage); },
                    key: 'PeoplePicker',
                    name: 'PeoplePicker',
                    url: '#/examples/PeoplePicker'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Persona/PersonaPage').PersonaPage); },
                    key: 'Persona',
                    name: 'Persona',
                    url: '#/examples/persona'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Pivot/PivotPage').PivotPage); },
                    key: 'Pivot',
                    name: 'Pivot',
                    url: '#/examples/pivot'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/ProgressIndicator/ProgressIndicatorPage').ProgressIndicatorPage); },
                    key: 'ProgressIndicator',
                    name: 'ProgressIndicator',
                    url: '#/examples/progressindicator'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Rating/RatingPage').RatingPage); },
                    key: 'Rating',
                    name: 'Rating',
                    url: '#/examples/rating'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/SearchBox/SearchBoxPage').SearchBoxPage); },
                    key: 'SearchBox',
                    name: 'SearchBox',
                    url: '#/examples/searchbox'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Spinner/SpinnerPage').SpinnerPage); },
                    key: 'Spinner',
                    name: 'Spinner',
                    url: '#/examples/spinner'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/TeachingBubble/TeachingBubblePage').TeachingBubblePage); },
                    key: 'TeachingBubble',
                    name: 'TeachingBubble',
                    url: '#/examples/teachingbubble'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/TextField/TextFieldPage').TextFieldPage); },
                    key: 'TextField',
                    name: 'TextField',
                    url: '#/examples/textfield'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Toggle/TogglePage').TogglePage); },
                    key: 'Toggle',
                    name: 'Toggle',
                    url: '#/examples/toggle'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Tooltip/TooltipPage').TooltipPage); },
                    key: 'Tooltip',
                    name: 'Tooltip',
                    url: '#/examples/Tooltip'
                }
            ],
            name: 'Basic components'
        },
        {
            links: [
                {
                    getComponent: function (cb) { return cb(require('../components/ColorPicker/ColorPickerPage').ColorPickerPage); },
                    key: 'ColorPicker',
                    name: 'ColorPicker',
                    url: '#/examples/colorpicker'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/GroupedList/GroupedListPage').GroupedListPage); },
                    key: 'GroupedList',
                    name: 'GroupedList',
                    url: '#examples/groupedlist'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Image/ImagePage').ImagePage); },
                    key: 'Image',
                    name: 'Image',
                    url: '#/examples/image'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Layer/LayerPage').LayerPage); },
                    key: 'Layer',
                    name: 'Layer',
                    url: '#/examples/layer'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Nav/NavPage').NavPage); },
                    key: 'Nav',
                    name: 'Nav',
                    url: '#/examples/nav'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Slider/SliderPage').SliderPage); },
                    key: 'Slider',
                    name: 'Slider',
                    url: '#/examples/slider'
                }
            ],
            name: 'Extended components'
        },
        {
            links: [
                {
                    getComponent: function (cb) { return cb(require('../components/FocusTrapZone/FocusTrapZonePage').FocusTrapZonePage); },
                    key: 'FocusTrapZone',
                    name: 'FocusTrapZone',
                    url: '#examples/focustrapzone'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/FocusZone/FocusZonePage').FocusZonePage); },
                    key: 'FocusZone',
                    name: 'FocusZone',
                    url: '#examples/focuszone'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/MarqueeSelection/MarqueeSelectionPage').MarqueeSelectionPage); },
                    key: 'MarqueeSelection',
                    name: 'MarqueeSelection',
                    url: '#examples/marqueeselection'
                },
                {
                    getComponent: function (cb) { return cb(require('../utilities/selection/SelectionPage').SelectionPage); },
                    key: 'Selection',
                    name: 'Selection',
                    url: '#examples/selection'
                },
                {
                    getComponent: function (cb) { return cb(require('../components/Theme/ThemePage').ThemePage); },
                    key: 'Theme',
                    name: 'Themes',
                    url: '#examples/themes'
                }
            ],
            name: 'Utilities'
        }
    ],
    headerLinks: [
        {
            name: 'Getting started',
            url: '#/'
        },
        {
            name: 'Fabric',
            url: 'http://dev.office.com/fabric'
        },
        {
            name: 'Github',
            url: 'http://www.github.com/officedev'
        }
    ]
};
exports.App = function (props) { return React.createElement(example_app_base_1.App, __assign({ appDefinition: exports.AppDefinition }, props)); };
