// office-ui-fabric-react/lib/components/Button/DefaultButton/DefaultButton.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var BaseButton_1 = require("../BaseButton");
var DefaultButton_scss_1 = require("./DefaultButton.scss");
var DefaultButton = (function (_super) {
    __extends(DefaultButton, _super);
    function DefaultButton() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.classNames = {
            base: 'ms-Button',
            variant: 'ms-Button--default',
            icon: DefaultButton_scss_1.default.icon,
            isDisabled: DefaultButton_scss_1.default.isDisabled,
            isEnabled: DefaultButton_scss_1.default.isEnabled,
            label: DefaultButton_scss_1.default.label,
            root: DefaultButton_scss_1.default.root
        };
        return _this;
    }
    DefaultButton.prototype.onRenderDescription = function () { return null; };
    return DefaultButton;
}(BaseButton_1.BaseButton));
exports.DefaultButton = DefaultButton;
