// office-ui-fabric-react/lib/components/Button/PrimaryButton/PrimaryButton.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var BaseButton_1 = require("../BaseButton");
var PrimaryButton_scss_1 = require("./PrimaryButton.scss");
var PrimaryButton = (function (_super) {
    __extends(PrimaryButton, _super);
    function PrimaryButton() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.classNames = {
            base: 'ms-Button',
            variant: 'ms-Button--primary',
            icon: PrimaryButton_scss_1.default.icon,
            isDisabled: PrimaryButton_scss_1.default.isDisabled,
            isEnabled: PrimaryButton_scss_1.default.isEnabled,
            label: PrimaryButton_scss_1.default.label,
            root: PrimaryButton_scss_1.default.root
        };
        return _this;
    }
    PrimaryButton.prototype.onRenderDescription = function () { return null; };
    return PrimaryButton;
}(BaseButton_1.BaseButton));
exports.PrimaryButton = PrimaryButton;
