// office-ui-fabric-react/lib-amd/components/Button/CommandButton/CommandButton.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "../BaseButton", "./CommandButton.scss"], function (require, exports, BaseButton_1, CommandButton_scss_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var CommandButton = (function (_super) {
        __extends(CommandButton, _super);
        function CommandButton() {
            var _this = _super !== null && _super.apply(this, arguments) || this;
            _this.classNames = {
                base: 'ms-Button',
                variant: 'ms-Button--command',
                icon: CommandButton_scss_1.default.icon,
                isDisabled: CommandButton_scss_1.default.isDisabled,
                isEnabled: CommandButton_scss_1.default.isEnabled,
                label: CommandButton_scss_1.default.label,
                root: CommandButton_scss_1.default.root,
                flexContainer: CommandButton_scss_1.default.flexContainer
            };
            return _this;
        }
        CommandButton.prototype.onRenderDescription = function () { return null; };
        return CommandButton;
    }(BaseButton_1.BaseButton));
    exports.CommandButton = CommandButton;
});
