// office-ui-fabric-react/lib/components/Button/ContextualButtonPage.visualtest.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = require("./index");
/* tslint:disable:no-unused-variable */
var React = require("react");
/* tslint:enable:no-unused-variable */
var CommandButtonVPage = (function (_super) {
    __extends(CommandButtonVPage, _super);
    function CommandButtonVPage() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CommandButtonVPage.prototype.render = function () {
        return React.createElement("div", null,
            React.createElement("div", null,
                React.createElement(index_1.DefaultButton, { id: 'ContextualButton', disabled: false, icon: 'Add', text: 'New', menuProps: {
                        className: 'ContextualButtonMenu',
                        id: 'ContextualButtonMenu',
                        items: [
                            {
                                key: 'emailMessage',
                                name: 'Email message',
                                icon: 'Mail'
                            },
                            {
                                key: 'calendarEvent',
                                name: 'Calendar event',
                                icon: 'Calendar'
                            }
                        ]
                    } })),
            React.createElement("div", null,
                React.createElement(index_1.DefaultButton, { id: 'ContextualButtonDisabled', icon: 'Add', text: 'New', disabled: true })));
    };
    return CommandButtonVPage;
}(React.Component));
exports.default = CommandButtonVPage;
